libusb/descriptor.c

```diff
@@ -1218,7 +1229,7 @@ static int parse_iad_array(struct libusb_context *ctx,
 
 	// First pass: Iterate through desc list, count number of IADs
 	iad_array->length = 0;
-	while (consumed < size) {
+	while (consumed + 1 < size) {
```

Comment:
Prevents off-by-one heap overflow in the case that the amount consumed less than size, but by less than the descriptor we're about to read.